### PR TITLE
Turn off light dismiss for non-toplevel menus.

### LINKED ISSUE #5475: ContextMenu: Opening a sub-menu will prevent mouse interaction with the parent menu level.
**Describe the bug**

As in the title and in the attached video - if sub-menu shows up and you hover over an element in it then interaction with the parent level of items becomes impossible.

There is also another problem - moving mouse from one parent element to another becomes very laggy if sub-menu begins to open.

**Screenshots**

https://user-images.githubusercontent.com/2588062/107791642-799dd780-6d54-11eb-978e-fafe23bb34f3.mp4

**Desktop (please complete the following information):**

- OS: Windows
- Version: Current master

### PR DESCRIPTION
**Pull request by grokys**
#5295 enabled the light dismiss overlay for popups, which in turn broke submenu interactions (#5475). The fix seems to be simple: turn off light dismiss for non-toplevel menus. These menus will be in popups so this means their behavior will be reverted to that of before #5295.

Fixes #5475

### CODE REVIEW

**Review by maxkatz6 — COMMENTED** [review]
I was thinking to use whole parent popup as a OverlayInputPassThroughElement parameter instead of single parent menu item element. 
But this approach makes even more sense for me. 
One thing I want to test later, if light dismiss still works for Menu elements (for ContextMenu it should work in any way). 

Upd: I see, there is another popup in templates for "Menu > MenuItem" selector. It works good.

**Comment by maxkatz6** [comment]
We need similar changes for Default theme as well

**Comment by grokys** [comment]
@maxkatz6 good point - done.

File: src/Avalonia.Themes.Default/MenuItem.xaml [hunk 1]
```diff
@@ -59,7 +59,7 @@
                   Grid.Column="4"/>
             <Popup Name="PART_Popup"
                    PlacementMode="Right"
-                   IsLightDismissEnabled="True"
+                   IsLightDismissEnabled="False"
                    OverlayInputPassThroughElement="{Binding $parent[MenuItem]}"
```
Comment (maxkatz6) [inline src/Avalonia.Themes.Default/MenuItem.xaml:63]:
```suggestion
```

**Review by maxkatz6 — CHANGES_REQUESTED** [review]

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**